### PR TITLE
Remove libjpeg package from rimage

### PR DESCRIPTION
**Pull request by randhid**
We might want to remove libjpeg turbo completely soon. Turns out this particular package was just serving static image files, so I'll test with `image_file` cameras as well as the webcams on different builds. before i take this out of draft.

Open question - what was the rgbd converted doing in the jpeg libraries, we can keep the non-libjpeg version if needed.

To do:

- [x] Test on x86 Linux with Webcam.  No FFMPEG or libjpeg-turbo present in the build environment
- [x] Test on Raspberry Pi with webcam. No FFMPEG or libjpeg-turbo present in the build environment
- [x] Test on Mac with Webcam.  No FFMPEG or libjpeg-turbo present in the build environment
- [x]  Test that the `image_file` viam component on x86 shows up and has feature parity. 
- [x] Test that the `image_file` viam component shows up on Mac and has feature parity. 
- [x] Test that `image_file` viam component on arm64 shows up and has feature parity.

### CODE REVIEW

**Comment by randhid** [comment]
Mac tests:
Screenshot 2024-08-26 at 20 11 28

**Comment by randhid** [comment]
Tests on raspberry pi
Screenshot 2024-08-26 at 21 08 15

File: etc/ld_wrapper.sh [hunk 1]
```diff
@@ -31,7 +31,7 @@ fi
 STRIPPED_ARGS="-g -O2"
 
 # list of arguments to prefix with -Bstatic
-STATIC_ARGS="-lx264 -lnlopt -ljpeg -ltensorflowlite_c -lpigpio -lstdc++"
+STATIC_ARGS="-lx264 -lnlopt -ltensorflowlite_c -lpigpio -lstdc++"
```
Comment (randhid) [inline etc/ld_wrapper.sh:34]: cc @Otterverse @abe-winter

File: rimage/jpeg.go [hunk 2]
```diff
@@ -1,26 +0,0 @@
-//go:build no_cgo
-
-package rimage
```
Comment (randhid) [inline rimage/jpeg.go:3]: Checked that these aren't used in `rtsp` `app` or `goutils`. I am okay with keeping the non libjpeg one though @bhaney, if the convert to RGBA conversion is necessary, but it seems to be special cased to the jpeg file in `image_file` only.

File: etc/ld_wrapper.sh (hunk 1 above)
Comment (abe-winter) [reply etc/ld_wrapper.sh:34]: hooray fewer dependencies

**Review by bhaney — APPROVED** [review]

**Review by Otterverse — APPROVED** [review]
LGTM! One thing to note, I think libjpeg is still embedded (statically) within the static libs pre-compiled in the gostream/ folder. But that should be fine until if/when we care.

**Comment by randhid** [comment]
Test on x86, @hexbabe is a viam-server expected to run with sudo on linux x86 systems? That was the hold up
Screenshot 2024-08-28 at 12 44 22

**Comment by randhid** [comment]
> LGTM! One thing to note, I think libjpeg is still embedded (statically) within the static libs pre-compiled in the gostream/ folder. But that should be fine until if/when we care.

We are apparently [not using them](https://github.com/viamrobotics/rdk/pull/4265)

**Comment by hexbabe** [comment]
> Test on x86, @hexbabe is a viam-server expected to run with sudo on linux x86 systems? That was the hold up Screenshot 2024-08-28 at 12 44 22

I've also had to run it with sudo. I did not have the curiosity or bandwidth at the time to investigate why

**Pull request merged**